### PR TITLE
webpack: add missing uglifyjs plugin typings

### PR DESCRIPTION
**Pull request by mattlewis92**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: webpack/index.d.ts [hunk 1]
```diff
@@ -1020,7 +1020,18 @@ declare namespace webpack {
             new (preferEntry: boolean): Plugin;
         }
         interface UglifyJsPluginStatic {
-            new (options?: UglifyJS.MinifyOptions): Plugin;
+            new (
+              options?: UglifyJS.MinifyOptions,
+              compress?: boolean | UglifyJS.CompressorOptions,
```
Comment (bcherny) [inline webpack/index.d.ts]: Aren't `compress`, `mangle`, etc.  keys in `UglifyJS.MinifyOptions`? Looking at the [docs](https://webpack.github.io/docs/list-of-plugins.html#uglifyjsplugin),  I don't think these are passed in as params.

Comment (mattlewis92) [inline webpack/index.d.ts]: Doh, I mis-read the typings, PR updated 😄

File: webpack/index.d.ts [hunk 2]
```diff
@@ -472,6 +472,15 @@ declare namespace webpack {
         apply(thisArg: Webpack, ...args: any[]): void;
     }
 
+    interface UglifyPluginOptions extends UglifyJS.MinifyOptions {
+        beautify?: boolean;
+        comments?: boolean | RegExp | Function;
```
Comment (bcherny) [inline webpack/index.d.ts]: A function signature would be nice instead of `Function`

File: webpack/webpack-tests.ts [hunk 3]
```diff
@@ -289,6 +289,12 @@ plugin = new webpack.optimize.UglifyJsPlugin({
     }
 });
 plugin = new webpack.optimize.UglifyJsPlugin({
+  sourceMap: false,
+  comments: true,
+  beautify: true,
+  exclude: /node_modules/
+});
```
Comment (bcherny) [inline webpack/webpack-tests.ts]: Might want to add some more tests to make sure your new types are well covered.

**Review by bcherny — APPROVED** [review]

**Comment by bcherny** [comment]
👍

File: webpack/index.d.ts (hunk 2 above)
Comment (mattlewis92) [inline webpack/index.d.ts]: Good shout, done

File: webpack/webpack-tests.ts (hunk 3 above)
Comment (mattlewis92) [inline webpack/webpack-tests.ts]: Done

**Comment by mattlewis92** [comment]
@bcherny thanks a lot for reviewing 👍

**Pull request merged**